src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -35,15 +35,20 @@ namespace ripple {
 
 class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
 {
-  private:
+private:
     AccountID src_;
     AccountID dst_;
     Currency currency_;
-    bool isLast_ = false;
 
     // Charge transfer fees when the prev step redeems
     Step const* const prevStep_ = nullptr;
 
+    // When offerCrossing then ignore trust line quality.
+    bool const offerCrossing_;
```

Comment:
Let's consider an offer x-ing subclass, much like for bookstep. We'd have virtuals for `maxFlow`, `quality` and `qualities`, I think.